### DISCUSSION TITLE
Making a New Addon

### DISCUSSION THREAD

**Discussion by Wjplatformer** [ScratchAddons/ScratchAddons#4944] [Help] [answered]
## Currently, I am thinking of making a new addon
which is to change outdated navbar to new one when viewing mystuff, profile page and account settings.
![image](https://user-images.githubusercontent.com/82447031/182568573-0d1a53b2-a3fe-4e82-9560-56d5761c0207.png)
# to
![image](https://user-images.githubusercontent.com/82447031/182568678-f175797e-a4eb-465e-81ac-8fb3f0c18692.png)

<br>

### That would be Jolly well but it is my first time making my addon...
how do I navigate The Sea Of Scripts in @LLK and test my addon? Do I painstakingly search through the repo manually or is there a tool to do it for me? Do I fork @LLK scripts and transfer them to my PC to do a local test? How does the addon work as a separate script not in the LLK repo. Is there a particularly time-saving way of doing it? Or you guys are really good at doing all these?

**Comment by CST1229** [comment]
There's already the Scratch 2.0 -> 3.0 (`scratchr2`) addon, unless you mean SPECIFICALLY the navbar and nothing else.

**Reply by CST1229** [reply]
And, usually, addons do not fork or use code from Scratch, as they are essentially userscripts and not Scratch mods.

**Reply by Wjplatformer** [reply]
Oh, Thanks then! I guess I did not notice this addon 🤔

**Reply by WorldLanguages** [reply]
Yeah this is part of the scratchr2 addon, so it doesn't make sense to have it as a separated addon as well.

**Comment by Norbiros** [comment, marked as answer]
Read https://scratchaddons.com/docs/develop/getting-started/creating-an-addon/

**Reply by Wjplatformer** [reply]
Thanks! If I need to do a new addon, I'm sure to bookmark this!
